```
$ nox-review pr 10499
[...]
building path(s) ‘/nix/store/8v8ky4lmjql7n4z8mx32a60ghr2jh4df-pijul-0.1’
unpacking sources
unpacking source archive /nix/store/786mdm2lgjnjv0yjj6x7zhx0v1y752h8-fetchdarcs
source root is fetchdarcs
patching sources
configuring
no configure script, doing nothing
building
make flags: SHELL=/nix/store/av9ybf7c0dr2mscxs5ccp8506ckvnj3l-bash-4.3-p42/bin/bash
bash make_stubs
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
make_stubs: line 15: gcc: command not found
make_stubs: line 16: /tmp/pijul_build/tmp: No such file or directory
cc -fPIC -Wall -c -o lmdb_stubs.o lmdb_stubs.c
lmdb_stubs.c:40:3: warning: incompatible pointer to integer conversion initializing 'value' (aka 'long') with an expression of type 'MDB_env *' (aka 'struct MDB_env *') [-Wint-conversion]
  CAMLreturn(env);
  ^~~~~~~~~~~~~~~
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:227:28: note: expanded from macro 'CAMLreturn'
#define CAMLreturn(result) CAMLreturnT(value, result)
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:222:8: note: expanded from macro 'CAMLreturnT'
  type caml__temp_result = (result); \
       ^                   ~~~~~~~~
lmdb_stubs.c:47:3: error: non-void function 'caml_mdb_env_copy' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:78:3: warning: incompatible pointer to integer conversion initializing 'value' (aka 'long') with an expression of type 'MDB_txn *' (aka 'struct MDB_txn *') [-Wint-conversion]
  CAMLreturn(txn);
  ^~~~~~~~~~~~~~~
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:227:28: note: expanded from macro 'CAMLreturn'
#define CAMLreturn(result) CAMLreturnT(value, result)
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:222:8: note: expanded from macro 'CAMLreturnT'
  type caml__temp_result = (result); \
       ^                   ~~~~~~~~
lmdb_stubs.c:84:3: error: non-void function 'caml_mdb_txn_abort' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:91:3: error: non-void function 'caml_mdb_txn_commit' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:125:3: error: non-void function 'caml_mdb_put' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:173:3: error: non-void function 'caml_mdb_del' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:181:3: error: non-void function 'caml_mdb_dbi_close' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:188:3: error: non-void function 'caml_mdb_drop' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:193:3: error: non-void function 'caml_mdb_env_close' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:208:3: error: non-void function 'caml_mdb_env_set_maxdbs' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:225:3: error: non-void function 'caml_mdb_cursor_renew' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:230:3: error: non-void function 'caml_mdb_cursor_close' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:266:3: error: non-void function 'caml_mdb_cursor_del' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
lmdb_stubs.c:279:3: error: non-void function 'caml_mdb_cursor_put' should return a value [-Wreturn-type]
  CAMLreturn0;
  ^
/nix/store/k8f89nm9l4jh4zqpwaiqjfc3gffmb1m0-ocaml-4.01.0/include/caml/memory.h:218:3: note: expanded from macro 'CAMLreturn0'
  return; \
  ^
2 warnings and 13 errors generated.
Makefile:32: recipe for target 'lmdb_stubs.o' failed
make: *** [lmdb_stubs.o] Error 1
builder for ‘/nix/store/m4m3park0vpki37n3xd409533a3p80ga-pijul-0.1.drv’ failed with exit code 2
error: build of ‘/nix/store/m4m3park0vpki37n3xd409533a3p80ga-pijul-0.1.drv’ failed
The invocation of "nix-build -A lmdb -A pijul /Users/davidak/.nox/nixpkgs" failed
```

--- a/pkgs/development/libraries/lmdb/default.nix
+++ b/pkgs/development/libraries/lmdb/default.nix
@@ -1,6 +1,7 @@
 { stdenv, fetchzip }:
 
-stdenv.mkDerivation rec {
+let optional = stdenv.lib.optional;
+in stdenv.mkDerivation rec {
   name = "lmdb-${version}";
   version = "0.9.16";
 
@@ -11,7 +12,8 @@ stdenv.mkDerivation rec {
 
   postUnpack = "sourceRoot=\${sourceRoot}/libraries/liblmdb";
 
-  makeFlags = "prefix=$(out)";
+  makeFlags = ["prefix=$(out)"]
+              ++ optional stdenv.cc.isClang "CC=clang";
 
   doCheck = true;
   checkPhase = "make test";

